lib/collection/variable-changeset.js

```diff
@@ -0,0 +1,235 @@
+var _ = require('../util').lodash,
+    Property = require('./property').Property,
+
+    /**
+     * @private
+     * @const
+     * @type {Object}
+     */
+    INSTRUCTION = {
+        SET: 'set',
+        UNSET: 'unset'
+    },
+
+    /**
+     * @private
+     * @const
+     * @type {String}
+     */
+    POSTMAN_PROPERTY_NAME = '_postman_propertyName',
+
+    /**
+     * @private
+     * @const
+     * @type {String}
+     */
+    VARIABLE_SCOPE = 'VariableScope',
+
+    /**
+     * Checks if the instance is a VariableScope. Defined here to avoid importing VariableScope and causing a
+     * circular dependency.
+     *
+     * @private
+     * @param {*} obj the object that needs to be validated
+     */
+    isVariableScope = function (obj) {
```

Comment:
If diff is aware of scope, it should then depend on it and use the `isVariableScope` static function. Since that would be cyclic (though node would handle it owing to singleton nature), we should simply not hard-check VariableScope.